File: cateyes-python/examples/bytecode.py
# -*- coding: utf-8 -*-
from __future__ import print_function

import cateyes


system_session = cateyes.attach(0)
bytecode = system_session.compile_script(name="bytecode-example", source="""\
'use strict';

rpc.exports = {
  listThreads: function () {
    return Process.enumerateThreadsSync();
  }
};
""")

session = cateyes.attach("Twitter")
script = session.create_script_from_bytes(bytecode)
script.load()
api = script.exports
print("api.list_threads() =>", api.list_threads())

File: cateyes-node/examples/bytecode.js
'use strict';

const cateyes = require('..');

const processName = process.argv[2];

const source = `'use strict';

rpc.exports = {
  listThreads: function () {
    return Process.enumerateThreadsSync();
  }
};
`;

async function main() {
  const systemSession = await cateyes.attach(0);
  const bytecode = await systemSession.compileScript(source, {
    name: 'bytecode-example'
  });

  const session = await cateyes.attach(processName);
  const script = await session.createScriptFromBytes(bytecode);
  script.message.connect(message => {
    console.log('[*] Message:', message);
  });
  await script.load();

  console.log('[*] Called listThreads() =>', await script.exports.listThreads());

  await script.unload();
}

main()
  .catch(e => {
    console.error(e);
  });